src/IECoreHoudini/SOP_SceneCacheSource.cpp

```diff
@@ -321,85 +329,123 @@ void SOP_SceneCacheSource::loadObjects( const IECoreScene::SceneInterface *scene
 		currentPath += scene->name().string();
 	}
 
-	if ( scene->hasObject() && UT_String( currentPath ).multiMatch( params.shapeFilter ) && tagged( scene, params.tagFilter ) )
+	if ( UT_String( currentPath ).multiMatch( params.shapeFilter ) && tagged( scene, params.tagFilter ) )
 	{
-		std::string name = relativePath( scene, rootSize );
-
-		Imath::M44d currentTransform;
-		if ( space == Local )
+		// filter out visibility or inherited visibility
+		if ( evalInt( pVisibilityFilter.getToken(), 0, 0 ) || inheritedVisibility )
 		{
-			currentTransform = scene->readTransformAsMatrix( time );
-		}
-		else if ( space != Object )
-		{
-			currentTransform = transform;
-		}
+			if ( scene->hasAttribute( IECoreScene::SceneInterface::visibilityName ) )
```

Comment:
Here is where I think we can optimise based on `inheritedVisibility`. This could be ` inheritedVisibility && scene->hasAttribute...`. If the visibility has been turned off at an ancestor location, we don't need to consider the attributes at child locations (the first non-visible ancestor should prune the whole hierarchy below).

Actually, this is more than an optimisation. Without it, the `inheritedVisibility = ` line inside the block could turn visibility back _on_ after it was turned off - that would be a bug.

Reply:
this seems to be the behaviour I was after, thanks for the suggestion, I've updated the PR accordingly